### DISCUSSION TITLE
Updating App with 1024-bit signing key

### DISCUSSION THREAD

**Discussion by ldrly-grant** [airsdk/Adobe-Runtime-Support#810] [Q&A]
I currently have an existing app that uses a 1024-bit signing key.  I need to make updates to it but have run into the following problem:
The app is currently being packaged/signed using Java and adt.jar via an ant script - excerpt as follows:
```
        <java jar="${air.sdk}/lib/adt.jar" fork="true" failonerror="true" maxmemory="2048m">
            <arg line="-package"/>
            <arg line="-target ${params.target.android}"/>
            <arg line="-listen" if:true="${params.debug}"/>
			<arg line="-arch ${android.arch}"/>
            <arg line="-storetype pkcs12"/>
            <arg line="-keystore '${air.key}'"/>
            <arg line="-storepass ${air.pass}"/>
            <arg line="'${project.out}/${package.name}'"/>
            <arg line="'${project.temp}/${platform.config}'"/>

```
With the 1024-bit key, Java 7 is required.
I'm using the distriqt ANEs which have switched to androidx, which do not appear to build properly with Java 7.  (If I use Java 8, the packaging portion appears to work properly but the signing step fails due to lack of support in Java 8+ for 1024-bit signing keys.)

What is the best way to package and sign the app using the updated ANEs/androidx and the 1024-bit signing key?
Is there a way to do the packaging step in Java 8 and the signing step in Java 7?
What other options are there?

**Comment by marchbold** [comment]
Use v1.8.112 of Java and you can support older 1024bit certificates.
